proc/community/src/main/java/org/neo4j/gds/louvain/LouvainStreamProc.java

```diff
@@ -82,26 +79,50 @@ protected Stream<StreamResult> stream(ComputationResult<Louvain, LouvainResult,
         return runWithExceptionLogging("Graph streaming failed", () -> {
             Graph graph = computationResult.graph();
 
+            boolean consecutiveIds = computationResult
+                    .config()
+                    .consecutiveIds();
+
+            var nodeProperties = (computationResult.isGraphEmpty() || !consecutiveIds) ? null : nodeProperties((computationResult));
+
+            var louvain = computationResult.result();
+
             return LongStream
-                .range(0, graph.nodeCount())
-                .boxed()
-                .map((nodeId) -> {
-                    long[] communities = getCommunities(computationResult, nodeId);
-                    Optional<Long> communityId = getCommunityId(computationResult, nodeId);
-
-                    return communityId.map(id -> new StreamResult(
-                            graph.toOriginalNodeId(nodeId),
-                            communities,
-                            id
-                    )).orElse(null);
-                })
-                .filter(Objects::nonNull);
+                    .range(0, graph.nodeCount())
+                    .boxed()
+                    .filter(nodeId -> nodeProperties == null || nodeProperties.isValid(nodeId))
```

Comment:
NIT the nodeProperties == null check is independent of the nodeId, so I would push that filter up with its own if case.

Reply:
Regarding this, I wanted to use `consecutiveIds` in the filter but I noticed that the value of `nodeProperties` does not depend only on `consecutiveIds`

I tried to write 2 different if cases based on the nullability of `nodeProperties` but the outcome seemed "dirty" to me.